### PR TITLE
[DO-NOT-MERGE] For Jenkins test : [DROOLS-7006] Publish Antora docs as drools job

### PR DESCRIPTION
**Pull request by tkobayas**
This PR is just to share code with @radtriste for Jenkins test. The actual PR is https://github.com/kiegroup/drools/pull/4459

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

**Comment by radtriste** [comment]
@tkobayas you did no need a PR to kiegroup for that but fair enough, let me check that :)

File: .ci/jenkins/Jenkinsfile.promote [hunk 1]
```diff
@@ -61,38 +61,51 @@ pipeline {
                 script {
                     dir(getRepoName()) {
                         checkoutRepo()
-                        mergeAndPush(getDeployPrLink())
-                        tagLatest()
+                        //mergeAndPush(getDeployPrLink())
+                        //tagLatest()
```
Comment (tkobayas) [inline .ci/jenkins/Jenkinsfile.promote:71]: @radtriste wrote

> please comment https://github.com/kiegroup/drools/pull/4459/files#diff-47f29a39bd966d9d1cf2914a3aa9e6d04c2369bc94a59db763c0b16fafb8baa6R64-R65

But the link jumps to the diff page of the PR but it doesn't point at specific lines. From the last part `R64-R65`, you meant commenting-out these 2lines?

File: .ci/jenkins/Jenkinsfile.promote [hunk 2]
```diff
@@ -61,38 +61,51 @@ pipeline {
...
+
+        //                if (shouldDeployToRepository()) {
+        //                    runMavenDeploy()
+        //                } else {
+        //                    echo 'Testing environment and no specific deploy repository given => no deployment'
+        //                }
+        //            }
+        //        }
+        //    }
+        //}
```
Comment (tkobayas) [inline .ci/jenkins/Jenkinsfile.promote:114]: I also commented whole `stage('Set next snapshot version')` as suggested.

File: .ci/jenkins/Jenkinsfile.promote (hunk 1 above)
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.promote:71]: exactly !

File: .ci/jenkins/Jenkinsfile.promote [hunk 3]
```diff
@@ -61,38 +61,51 @@ pipeline {
                 script {
                     dir(getRepoName()) {
                         checkoutRepo()
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.promote:69]: in the `checkoutRepo` method, I would update that line: https://github.com/kiegroup/drools/blob/7d05384faaf89090a7086ff7248822246fe4c4db/.ci/jenkins/Jenkinsfile.promote#L227 and replace:
`getBuildBranch()` -> `getGitTag()`
for testing.

That way we are sure to take the 8.27.0.Final tag in our testing and upload content for that tag.

File: .ci/jenkins/Jenkinsfile.promote (hunk 1 above)
Comment (tkobayas) [reply .ci/jenkins/Jenkinsfile.promote:71]: Thanks!

File: .ci/jenkins/Jenkinsfile.promote (hunk 3 above)
Comment (tkobayas) [reply .ci/jenkins/Jenkinsfile.promote:69]: Thanks! Updated.

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#477` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/477/display/redirect

**Test results:**
- PASSED: 1912
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/477/testReport/org.kie.kogito.taskassigning.service/SolverExecutorTest/stopWithSolverStarted/">org.kie.kogito.taskassigning.service.SolverExecutorTest.stopWithSolverStarted</a></summary>
java.util.concurrent.TimeoutException: stopWithSolverStarted() timed out after 50 seconds<br/>	at org.junit.jupiter.engine.extension.TimeoutExceptionFactory.create(TimeoutExceptionFactory.java:29)<br/>	at org.junit.jupiter.engine.extension.SameThreadTimeoutInvocation.proceed(SameThreadTimeoutInvocation.java:58)<br/>	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)<br/>	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)<br/>	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)<br/>	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)<br/>	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)<br/>	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)<br/>	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)<br/>	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)<br/>	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)<br/>	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)<br/>	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)<br/>	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)<br/>	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)<br/>	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)<br/>	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)<br/>	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)<br/>	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)<br/>	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)<br/>	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)<br/>	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)<br/>	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)<br/>	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)<br/>	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)<br/>	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)<br/>	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)<br/>	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)<br/>	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)<br/>	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)<br/>	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)<br/>	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)<br/>	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)<br/>	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)<br/>	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)<br/>	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)<br/>	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)<br/>	Suppressed: java.lang.InterruptedException<br/>		at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:385)<br/>		at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)<br/>		at org.kie.kogito.taskassigning.service.SolverExecutorTest.stopWithSolverStarted(SolverExecutorTest.java:133)<br/>		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/>		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>		at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>		at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)<br/>		at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)<br/>		at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)<br/>		at org.junit.jupiter.engine.extension.SameThreadTimeoutInvocation.proceed(SameThreadTimeoutInvocation.java:45)<br/>		... 64 more<br/>
</details>

**Pull request closed without merging**